pkg/machines/components/createVmDialog.jsx

```diff
@@ -0,0 +1,435 @@
+                });
+                notifyValuesChanged('os', os);
+                break;
+            }
+            case 'os':
+            case 'source':
+                this.setState({ [key]: value });
+                if (valueParams) {
+                    notifyValuesChanged('error', valueParams.error);
+                }
+                break;
+            case 'sourceType':
+                this.setState({
+                    [key]: value,
+                    'source': null,
+                });
+                notifyValuesChanged('source', null);
+                break;
+            case 'memorySize':
+                this.setState({ [key]: value });
+                value = convertToUnit(value, this.state.memorySizeUnit, units.MiB);
+                break;
+            case 'storageSize':
+                this.setState({ [key]: value });
+                value = convertToUnit(value, this.state.storageSizeUnit, units.GiB);
+                break;
+            case 'memorySizeUnit':
+                this.setState({ [key]: value });
+                value = convertToUnit(this.state.memorySize, value, units.MiB);
+                key = 'memorySize';
+                break;
+            case 'storageSizeUnit':
+                this.setState({ [key]: value });
+                value = convertToUnit(this.state.storageSize, value, units.GiB);
+                key = 'storageSize';
+                break;
+            default:
+                break;
+        }
+
+        notifyValuesChanged(key, value);
+    }
+
+    render() {
+        const vendorSelectEntries = [];
+
+        if (this.props.familyMap[DIVIDER_FAMILY]) {
+            vendorSelectEntries.push((
+                <Select.SelectEntry data={NOT_SPECIFIED} key={NOT_SPECIFIED}>{_(NOT_SPECIFIED)}</Select.SelectEntry>));
+            vendorSelectEntries.push((<Select.SelectDivider/>));
+        }
+
+        this.props.familyList.forEach(({ family, vendors }) => {
+            if (family === DIVIDER_FAMILY) {
+                return;
+            }
+            vendorSelectEntries.push((<Select.SelectHeader>{family}</Select.SelectHeader>));
+
+            vendors.forEach((vendor) => {
+                vendorSelectEntries.push((
+                    <Select.SelectEntry data={vendor} key={vendor}>{vendor}</Select.SelectEntry>));
+            });
+        });
+
+        const osEntries = (
+            this.props.vendorMap[this.state.vendor]
+                .map(os => (<Select.SelectEntry data={os.shortId}
+                                                key={os.shortId}>{getOSStringRepresentation(os)}</Select.SelectEntry>))
+        );
+
+        let installationSource;
+        switch (this.state.sourceType) {
+            case COCKPIT_FILESYSTEM_SOURCE:
+                installationSource = (
+                    <FileAutoComplete.FileAutoComplete id="source"
+                                                       placeholder={_("Path to Cockpit's local file")}
```

Comment:
I think this might be ambiguous.
@andreasn , what about 
`Path to ISO file on host's file system`
?

Reply:
I agree. We shouldn't refer to "Cockpit" anywhere in the UI. It's RHEL/Fedora/Debian as far as the user is concerned.